plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcTypeConversions.java

```diff
@@ -31,13 +35,34 @@ private OrcTypeConversions() {}
     public static Function<Block, Block> getTypeConversion(OrcTypeKind orcType, Type trinoType)
     {
         if (orcType == BYTE) {
+            if (trinoType == SMALLINT) {
+                return OrcTypeConversions::tinyintToSmallint;
+            }
             if (trinoType == INTEGER) {
                 return OrcTypeConversions::tinyintToInteger;
             }
+            if (trinoType == BIGINT) {
+                return OrcTypeConversions::tinyintToBigint;
+            }
         }
         return block -> block;
     }
 
+    private static Block tinyintToSmallint(Block block)
+    {
+        boolean[] isNull = new boolean[block.getPositionCount()];
+        short[] values = new short[block.getPositionCount()];
+        for (int i = 0; i < block.getPositionCount(); i++) {
+            if (block.isNull(i)) {
```

Comment:
if this had to be of top performance, this probably would want to have a simpler loop when `! block.mayHaveNull`.

no change requested, since I don't know the perf expectations

Reply:
BTW would it work to avoid the `if` here?

```java
isNull[i] = block.isNull(i)
values[i] = block.getByte(i, 0);
```


cc @dain @sopel39 @raunaqmorarka may know if it is legit

Reply:
I don't get what was gained by adding abstractions at page source layer for these column adaptations rather than letting the column reader handle it as it does today.
My expectation is that there should be no perf regression from the existing approach (demonstrated through JMH results).
Preserving the `mayHaveNull` of the original block is definitely required, downstream operators take advantage of that to improve performance.
Avoiding branch in the above way looks okay to me, however the contract in Block doesn't explcitly specify any expected behaviour for getter on null positions. So I don't know if that can be safely relied on or in future it may change to throw an exception.